compiler/rustc_codegen_llvm/src/context.rs

```diff
@@ -92,6 +97,15 @@ pub struct CodegenCx<'ll, 'tcx> {
     local_gen_sym_counter: Cell<usize>,
 }
 
+pub struct TypeLowering<'ll> {
+    /// Associated LLVM type
+    pub lltype: &'ll Type,
+
+    /// If padding is used the slice maps fields from source order
+    /// to llvm order.
+    pub field_remapping: Option<Box<SmallVec<[u32; 4]>>>,
```

Comment:
Another thing you can try (after this perf run) is removing the `Box` - the size of `TypeLowering` might not matter as much as not introducing additional allocations in the case where `field_remapping` is needed.